portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,43 +37,53 @@
 public class DataServlet extends HttpServlet {
 
     DatastoreService datastore;
+    UserService userService;
 
     @Override
     public void init(){
         datastore = DatastoreServiceFactory.getDatastoreService();
+        userService = UserServiceFactory.getUserService();
     }
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Query query = new Query(ServletConstants.COMMENT_ENTITY_ID).addSort(ServletConstants.COMMENT_TIMESTAMP_ID, SortDirection.DESCENDING);
-        PreparedQuery results = datastore.prepare(query);
+        Query query = new Query(ServletConstants.COMMENT_ENTITY_ID).addSort(ServletConstants.ENTITY_TIMESTAMP_ID, SortDirection.ASCENDING);
+        response.setContentType("text/html;");
 
-        ArrayList<String> messages = new ArrayList<>();
-        for(Entity entity : results.asIterable()){
-            messages.add((String)entity.getProperty(ServletConstants.COMMENT_TEXT_ID));
+        PreparedQuery results = datastore.prepare(query);
+        
+        if(userService.isUserLoggedIn()){    
+            ArrayList<ArrayList<String>> userComments = new ArrayList<>();
+            for(Entity entity : results.asIterable()){
+                ArrayList<String> userCommentCombo = new ArrayList<String>();
+                userCommentCombo.add((String)entity.getProperty(ServletConstants.ENTITY_EMAIL_ID));
+                userCommentCombo.add((String)entity.getProperty(ServletConstants.ENTITY_COMMENT_ID));
```

Comment:
Conveying semantic information (email id vs comment body) by positional information in an array isn't great from a readability and maintainability perspective. Ideally these would be stored as a structured object of some form, probably a value class with clearly-named accessors. This value class could also then handle the formatting of each individual comment into JSON